### ISSUE TITLE
Error when adding Command Set customizer to the document library

### ISSUE THREAD

**Issue by AJIXuMuK** [SharePoint/sp-dev-docs#1757]
#### Category
- [x] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

What could be the case for such an issue when adding a Command Set customizer (both in debug or by adding the app):
```
TypeError: Cannot read property 'toString' of undefined
    at r (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/splistdeferred-mini-9d1aa9ce.js:1)
    at C (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/splistdeferred-mini-9d1aa9ce.js:1)
    at spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/splistdeferred-mini-9d1aa9ce.js:1
    at Object.C [as _notify] (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spofiles-mini-311dff32.js:1)
    at Object.enter (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spofiles-mini-311dff32.js:1)
    at d._run (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spofiles-mini-311dff32.js:1)
    at d._completed (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spofiles-mini-311dff32.js:1)
    at n.e.onApplicationLoaded (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spoapp-mini-d9b3321e.js:1)
    at r (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spoapp-mini-d9b3321e.js:1)
    at window._spLoaderCallback (spoprod-a.akamaihd.net/files/odsp-next-prod_2018-04-13-sts_20180424.001/spoapp-mini-d9b3321e.js:1)
```
Some details:
1. Tested on simple List and Doc Library (no additional columns)
2. There are no custom actions applied to library, web, or site
3. The site where tested is a subsite of a site collection with publishing enabled. **BUT** I've tried to deactivate publishing and activate Site Pages feature. The same result.
4. Master page - Seattle
5. Modern UI is disabled by default for the tenant but enabled for the Document Library in Advanced Settings
6. The Modern Doc Library Page looks fine. If I remove the Command Set the console output after page refresh looks exactly the same as on working site.
7. And yes, I've tested the Command Set on other sites and it works.

So, basically my understanding is that there is some customization or some setting that is not configured.
The reason for this question is try to get the direction as the error is not informative at all.

Thanks!

**Comment by patmill** [comment]
If you break on exceptions in the browser debugger, what does the offending code look like during the toString on an undefined object?

**Comment by AJIXuMuK** [comment]
Hi @patmill ,
Thanks for spending time on that one!

So, it fails in `splistdeferred` while processing the list of custom actions.
```typescript
var s = a.clientSideComponentId.toString();
```

The object looks like that:
```json
{
    "text": "K2 Workflows",
    "iconUrl": null,
    "action": "javascript:LaunchApp('b8ac084a-243d-4e7a-988b-f145d47a352d', 'i:0i.t|ms.sp.ext|10d4f504-a2e2-4336-82cd-2a44382e35c6@8395b8e4-d017-486b-a04e-7e46b2103d4b', 'https:\\u002f\\u002fCreativedc-f32fa6ec3d7608.sharepoint.com\\u002fsites\\u002fapps\\u002fK2AppitforSharePoint\\u002fPages\\u002fApplication.aspx?formUrl=\\u002fRuntime\\u002fForm\\u002fK2\\u002bWorkflow\\u002bManual\\u00253FSPSiteURL\\u00253D{HostUrl}\\u0026SPListId\\u00253D{ListId}\\u0026SPItemId\\u00253D{ItemId}\\u0026SPItemUrl\\u00253D{ItemUrl}\\u0026Source\\u00253D{Source}', null);"
}
```

So, here the issue is that there is a 'K2 Workflows' action (weirdly that it was not included in the list of Custom Actions I got via script...). And I believe it's a classic ECB CustomAction.

I can ask the customer to disable this 3d party solution. But it would be great if such solutions don't stop the processing of others :)

Please, let me know what you think about it before closing the Issue.
Thanks!

**Comment by patmill** [comment]
Perfect - that's super helpful.  It looks like an issue in the logic that processes the custom actions where it first tries to get the component ID and then checks if it is valid, but it can't.

A possible work around if you need to get something going right away (it may or may not work) would be to update the SPUserCustomAction of the K2 workflow app and give the configuration a clientSideComponentId of the empty GUID.

In the mean time I'll follow up with the owner on a fix.

**Comment by AJIXuMuK** [comment]
Thanks @patmill !
Do you want me to close this one ore convert into "bug"?

**Comment by darrenparkinson** [comment]
Hi, I'm getting a similar error when trying to load a ListView Command Set.  Neither of my apps are showing in the top bar -- any thoughts how I might fix that?  Thanks.

`TypeError: Cannot read property 'toString' of undefined
    at r (splistdeferred-mini-66af9096.js:1)
    at C (splistdeferred-mini-66af9096.js:1)
    at splistdeferred-mini-66af9096.js:1
    at r.then (sp-http-mini-6ef881a8.js:1)
    at e.loadCustomActionExtensions (splistdeferred-mini-66af9096.js:1)
    at splistdeferred-mini-66af9096.js:1
    at Object.C [as _notify] (sp-http-mini-6ef881a8.js:1)
    at Object.enter (sp-http-mini-6ef881a8.js:1)
    at f._run (sp-http-mini-6ef881a8.js:1)
    at f._completed (sp-http-mini-6ef881a8.js:1)`

**Comment by patmill** [comment]
Missed closing this a long time ago.  @darrenparkinson - I'm assuming that yours has been addressed, but if not, please open a new issue.
